### PR TITLE
Split dataset, calc overlap_features, build vocab

### PR DESCRIPTION
**Pull request by hatianzhang**
1. Split xml into different folders.
2. Calc overlap features.
3. Build vocab.
4. Upd README

### CODE REVIEW

File: TrecQA/overlap_features.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  punct = set(string.punctuation)
+  stoplist.update(punct) 
+  #stoplist = None 
+ 
+  all_questions, all_answers, all_qids = [], [], []
+  base_dir = './'
+  # base_dir = '../../data/' + sys.argv[1] + '/'
+
+  sub_dirs = ['train/','train-all/', 'raw-dev/', 'raw-test/']
+
```
Comment (lintool) [inline TrecQA/overlap_features.py:130]: why raw-dev and raw-test instead of just dev and test?

File: TrecQA/stopwords.txt [hunk 2]
```diff
@@ -0,0 +1,238 @@
+'s
```
Comment (lintool) [inline TrecQA/stopwords.txt:1]: where does this stoplist come from?

File: TrecQA/overlap_features.py (hunk 1 above)
Comment (hatianzhang) [reply TrecQA/overlap_features.py:130]: TrecQA also has clean-dev and clean-test. What we generate is raw-dev and raw-test

File: TrecQA/stopwords.txt (hunk 2 above)
Comment (hatianzhang) [reply TrecQA/stopwords.txt:1]: From our own sm repo, I will ask jinfeng about it.

File: TrecQA/overlap_features.py (hunk 1 above)
Comment (lintool) [reply TrecQA/overlap_features.py:130]: okay, so what's clean-dev and clean-test then?

Comment (hatianzhang) [reply TrecQA/overlap_features.py:130]: The Clean version (Wang and Ittycheriah et al. 2015, Tan et al. 2015, dos Santos et al. 2016, Wang et al. 2016) removed questions with no answers or with only positive/negative answers, thus has only 65 questions in the development set and 68 questions in the test set. From ACL QA wiki page.

Comment (lintool) [reply TrecQA/overlap_features.py:130]: Ugh, seriously? :(
That sounds like bad experimental practice...
Man, we really need a brand-new QA dataset.

Comment (hatianzhang) [reply TrecQA/overlap_features.py:130]: In fact, we can do all own. We now have all documents and sentences which can easily generate a new test and dev set.

Comment (lintool) [reply TrecQA/overlap_features.py:130]: http://www.aclweb.org/aclwiki/index.php?title=Question_Answering_(State_of_the_art)

So there's actually four different conditions: {TRAIN, TRAIN-ALL} x {clean, raw}?

Comment (hatianzhang) [reply TrecQA/overlap_features.py:130]: Over time, the original dataset diverged to two versions due to different pre-processing in recent publications: both have the same training set but their development and test sets differ.   Train/Train-all remain the same. No clean version.

**Comment by gauravbaruah** [comment]
Building the dataset for the SMModel using this PR works!

**Review by gauravbaruah — APPROVED** [review]
